[Feat] 공통 컴포넌트 Modal 구현

kamja0510 opened the pull request:
## Related issue 🛠
- closed #27 

## Work Description ✏️
- [x] SingleActionModal 구현
- [x] DualActionModal 구현
- [x] Modal이 스크린에서 어떻게 쓰일지 구현

## Screenshot 📸
<table width="100%">
    <tr>
      <td width="40%">설명</td>
      <td width="60%">스크린샷</td>
    </tr>
    <tr>
      <td width="40%">
        SingleActionModal UI
      </td>
      <td width="60%">
        <img src="https://github.com/user-attachments/assets/4a13f75f-1476-4c37-9fca-38d69e1ab19e" style="height:auto;">
      </td>
    </tr>
    <tr>
      <td width="40%">DualActionModal UI</td>
      <td width="60%">
        <img src="https://github.com/user-attachments/assets/06132ab7-eed3-40e5-880d-431649a96128" style="height:auto;">
      </td>
    </tr>
    <tr>
      <td width="40%">ModalInteraction</td>
      <td width="60%">
        <video src="https://github.com/user-attachments/assets/05e495ad-bac6-4913-b5ee-26254d6aa489">
      </td>
    </tr>
</table>

## To Reviewers 📢
커스텀 모달을 이런식으로 사용하는게 맞을까요? 스크린에서 사용할 때 고려사항을 생각해서 프리뷰 구현해봤는데

1. 백핸들러(뒤로가기 버튼 클릭시 모달이 닫혀야함)
2. 전체 스크린에 Box 
a. 반투명 필터 걸어서 scrim 효과 줌
b. 뒷 배경 클릭시 onDismiss 효과 내도록 함
3. 모달이 자연스럽게 나타나도록 애니메이션 걸어줌

더 고려해야 할 사항이 있을까요?

kamja0510 committed:
[chore] #27 모달 관련 문자열 추출

kamja0510 committed:
[chore] #27 모달에 쓰이는 키 이미지 추가

kamja0510 committed:
[feat] #27 noRippleClickable 추가

kamja0510 committed:
[feat] #27 모달용 버튼 구현

kamja0510 committed:
[feat] #27 SingleActionModal.kt 구현

kamja0510 committed:
[feat] #27 DualActionModal.kt 구현

kamja0510 committed:
[chore] #27 ktlint

t1nm1ksun commented on core/designsystem/src/main/java/com/example/designsystem/component/button/ModalButton.kt:
```diff
@@ -0,0 +1,31 @@
+            )
+            .noRippleClickable(onClick = onClick)
+            .padding(vertical = 16.dp),
+        contentAlignment = Alignment.Center
```
근데 저는 뭔가 이 속성값들을 바로 ActionRow내부 텍스트들에 붙여준다면 불필요한 레이아웃 중첩은 방지할 수 있을 것 같은데 어떻게 생각하시나요??

kamja0510 replied:
맞는말입니다!
중복으로 여러번 작성하지 않으려고 추출해놓은 것이기 때문입니다.

t1nm1ksun reviewed (commented):
<img width="240" height="160" alt="image" src="https://github.com/user-attachments/assets/b512e51f-5f6c-4ca1-9747-8a27bdd31969" />

여기두 피그마 네이밍 권장!

t1nm1ksun reviewed (commented):
근데 디자인 확인해보니 그림자 넣어야 될 것 같아요!

kamja0510 committed:
[merge] #27 branch 'develop' into feat/#27-modal

# Conflicts:
#	core/designsystem/src/main/java/com/example/designsystem/util/ModifierExt.kt

kamja0510 committed:
[chore] #27 string resource 오타 수정

kamja0510 committed:
[chore] #27 SingleActionModal.kt -> GraphicModal.kt 네이밍 변경

kamja0510 committed:
[chore] #27 DualActionModal.kt -> WarningModal.kt 네이밍 변경

kamja0510 committed:
[feat] #27 WarningModal에 dropShadow추가

kamja0510 committed:
[refactor] #27 DPlayBaseIcon 사용

kamja0510 committed:
[merge] #27 branch 'develop' into feat/#27-modal

# Conflicts:
#	core/designsystem/src/main/java/com/example/designsystem/util/ModifierExt.kt
#	core/designsystem/src/main/res/values/strings.xml

kamja0510 committed:
[chore] #27 ktlint
